Helper/Inventory.php

```diff
@@ -126,11 +127,37 @@ private function getQuantityAndStockStatusWithMSI(ProductModel $product, ?int $s
     {
         $stockItemData = $this->getStockItemData($product->getSku(), $stockId);
         $qty = $stockItemData[GetStockItemDataInterface::QUANTITY];
-        $availability = $stockItemData[GetStockItemDataInterface::IS_SALABLE];
+        $availability = $this->isProductSalable($product, $stockId);
 
         return [$qty, $availability];
     }
 
+    /**
+     * Get info about the salability of a product
+     * If a product is a grouped product, we consider it is salable
+     * if any of its associated products is salable
+     *
+     * @param ProductModel $product
+     * @param int|null $stockId
+     *
+     * @return boolean
+     */
+    private function isProductSalable(ProductModel $product, ?int $stockId = null)
+    {   
+        if ($product->getTypeId() == Grouped::TYPE_CODE) {
+            $associatedProducts = $product->getTypeInstance()->getAssociatedProducts($product);
+            foreach ($associatedProducts as $associatedProduct) {
+                if ($this->isProductSalable($associatedProduct, $stockId)) {
```

Comment:
This recursion could be a bit confusing, I would have split it into two separated functions, but it should be valid this way too, so it's up to you :raised_hands: